src/App.tsx

```diff
@@ -1,26 +1,221 @@
+  };
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterOption) {
+        case FilterOptions.Active:
+          return !todo.completed;
+        case FilterOptions.Completed:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  }, [todos, filterOption]);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      setIsLoading(true);
+      try {
+        const todosFromApi = await getTodos();
+
+        setTodos(todosFromApi);
+      } catch {
+        setError('Unable to load todos');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      const timeout = setTimeout(() => setError(''), 3000);
+
+      return () => clearTimeout(timeout);
+    }
+  }, [error]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const toggleTodoCompletion = useCallback((id: number) => {
+    setTodos(prevTodos =>
+      prevTodos.map(todo =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+      ),
+    );
+  }, []);
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={`todoapp__toggle-all ${allCompleted ? 'active' : ''}`}
+            onClick={toggleAll}
+            aria-label="Toggle all todos"
+          />
+          <form onSubmit={handleAddTodo}>
+            <input
+              value={newTodoTitle}
+              onChange={e => setNewTodoTitle(e.target.value)}
+              type="text"
+              data-cy="NewTodoField"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              disabled={isLoading}
+              ref={inputRef}
+            />
+          </form>
+        </header>
+
+        {tempTodo && (
+          <div className="todo temp-todo">
+            <span>{tempTodo.title}</span>
+            <div className="loader" />
+          </div>
+        )}
+
+        <TodoList
+          todos={filteredTodos}
+          onToggleTodo={toggleTodoCompletion}
+          onDelete={handleDelete}
+        />
+
+        <Footer
+          todos={todos}
+          filterOption={filterOption}
+          setFilterOption={setFilterOption}
+          handleClearCompleted={handleClearCompleted}
+          isLoading={false}
+        />
```

Comment:
current failed test telling do not show footer without any todos.
![Screenshot 2024-12-26 at 14 36 51](https://github.com/user-attachments/assets/89e345e5-2a69-4d35-b4ea-c41a53c08d80)